org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/JavaSearchScope.java

```diff
@@ -455,15 +458,10 @@ private boolean encloses(String enclosingPath, String path, int index) {
 @Override
 public boolean encloses(IJavaElement element) {
 	if (this.elements != null) {
-		for (int i = 0, length = this.elements.size(); i < length; i++) {
-			IJavaElement scopeElement = (IJavaElement)this.elements.get(i);
-			IJavaElement searchedElement = element;
-			while (searchedElement != null) {
-				if (searchedElement.equals(scopeElement))
-					return true;
-				searchedElement = searchedElement.getParent();
-			}
-		}
+		do {
+			if (this.elementTypes[element.getElementType()] && this.elements.contains(element))
+				return true;
+		} while ((element = element.getParent()) != null);
```

Comment:
that line changes "element" instead of temporary "searchedElement" copy. element is used later

Reply:
`element` is modified in a branch that always returns so there is no impact to the lower brach.